analyses/rnaseq-batch-correct/code/01-ruvseq-deseq.R

```diff
@@ -0,0 +1,373 @@
+    opt_str = "--cohort_values",
+    type = "character",
+    help = "cohort value"
+  ),
+  make_option(
+    opt_str = "--k_value",
+    type = "character",
+    help = "number of K values to evaluate"
+  ),
+  make_option(
+    opt_str = "--pos_c",
+    type = "character",
+    help = "path to file of positive control genes"
+  ),
+  make_option(
+    opt_str = "--neg_c",
+    type = "character",
+    help = "path to file of negative control genes"
+  )
+)
+
+# extract parameters
+opt <- parse_args(OptionParser(option_list = option_list))
+dataset <- opt$dataset
+cancer_group_values <-
+  unlist(stringr::str_split(opt$cancer_group_values, ','))
+cohort_values <- opt$cohort_values
+k_value <- as.numeric(opt$k_value)
+neg_ctrl_genes.f <- opt$neg_c
+pos_ctrl_genes.f <- opt$pos_c
+
+# establish directories
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+analysis_dir <-
+  file.path(root_dir, 'analyses', 'rnaseq-batch-correct')
+input_dir <- file.path(analysis_dir, 'input')
+plots_dir <- file.path(analysis_dir, 'plots', dataset)
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir, recursive = TRUE)
+}
+
+output_dir <- file.path(analysis_dir, 'output', dataset)
+if (!dir.exists(output_dir)) {
+  dir.create(output_dir, recursive = TRUE)
+}
+
+# source functions
+source('util/deseq2_pvals_histogram.R') # DESeq2 pval histograms
+source('util/edaseq_plot.R') # PCA and UMAP clustering
+source('util/ruvg_test.R') # function to run RUVg
+
+# read histology
+message("Reading in histologies file")
+htl_df <- readr::read_tsv('../../data/v10/histologies.tsv')
+
+# read expected counts
+message("Reading RSEM expected counts file")
+cnt_df <-
+  readRDS('../../data/v10/gene-counts-rsem-expected_count-collapsed.rds')
+
+# filter histology
+selected_htl_df <- htl_df %>%
+  dplyr::filter(
+    experimental_strategy == "RNA-Seq",
+    sample_type == "Tumor",
+    cancer_group %in% cancer_group_values,!is.na(molecular_subtype)
+  )
+
+# filter expression
+cnt_df <- cnt_df %>%
+  dplyr::select(selected_htl_df$Kids_First_Biospecimen_ID)
+cnt_df <- cnt_df[rowSums(cnt_df) > 0, ]
+
+# filter by expression and convert to DGElist
+bs_id <-
+  gsub('TARGET-[0-9]{2}-',
+       '',
+       selected_htl_df$Kids_First_Biospecimen_ID) # shorten bs ids for TARGET samples
+molecular_subtype <-
+  factor(as.character(selected_htl_df$molecular_subtype))
+RNA_library <- factor(as.character(selected_htl_df$RNA_library))
+design <- model.matrix(~ molecular_subtype)
+counts_object <-
+  edgeR::DGEList(counts = cnt_df, group = molecular_subtype)
+counts_object_filtered <-
+  edgeR::filterByExpr(counts_object, group = molecular_subtype)
+counts_object <-
+  counts_object[counts_object_filtered, , keep.lib.sizes = FALSE]
+
+# create new expression set
+seq_expr_set <-
+  EDASeq::newSeqExpressionSet(
+    counts = round(counts_object$counts),
+    phenoData = data.frame(bs_id, molecular_subtype, RNA_library, row.names = colnames(round(
+      counts_object$counts
```

Comment:
```suggestion
    phenoData = data.frame(bs_id, factor(as.character(selected_htl_df$molecular_subtype)), factor(as.character(selected_htl_df$RNA_library)), row.names = colnames(round(
      counts_object$counts
```
Would this work? Then you can remove lines 108 & 109. Would also have to change in other places.

Reply:
i hesitate to change given that this is used in other parts of the script, namely to build the deseq2 dataset as well as the seq expression set for non-batch DGE analysis and plotting, respectively.